geotribu_cli/content/header_check.py

```diff
@@ -0,0 +1,234 @@
+import os
+from pathlib import Path
+
+import frontmatter
+from unidecode import unidecode
+
+from geotribu_cli.constants import GeotribuDefaults
+from geotribu_cli.json.json_client import JsonFeedClient
+from geotribu_cli.utils.check_image_size import get_image_dimensions_by_url
+from geotribu_cli.utils.check_path import check_path_exists
+
+logger = logging.getLogger(__name__)
+defaults_settings = GeotribuDefaults()
+
+MANDATORY_KEYS = [
+    "title",
+    "authors",
+    "categories",
+    "date",
+    "description",
+    "license",
+    "tags",
+]
+
+# ############################################################################
+# ########## CLI #################
+# ################################
+
+
+def parser_header_check(
+    subparser: argparse.ArgumentParser,
+) -> argparse.ArgumentParser:
+    """Set the argument parser subcommand.
+
+    Args:
+        subparser (argparse.ArgumentParser): parser to set up
+
+    Returns:
+        argparse.ArgumentParser: parser ready to use
+    """
+    subparser.add_argument(
+        "content_path",
+        help="Chemin du fichier markdown dont l'entête est à vérifier",
+        type=str,
+        metavar="content",
+        nargs="+",
+    )
+    subparser.add_argument(
+        "-af",
+        "--authors-folder",
+        dest="authors_folder",
+        type=Path,
+        help="Chemin qui contient les presentations markdown des auteurs/autrices",
+    )
+    subparser.add_argument(
+        "-minw",
+        "--min-width",
+        dest="min_image_width",
+        default=400,
+        type=int,
+        help="Largeur minimum de l'image à vérifier",
+    )
+    subparser.add_argument(
+        "-maxw",
+        "--max-width",
+        dest="max_image_width",
+        default=800,
+        type=int,
+        help="Largeur maximum de l'image à vérifier",
+    )
+    subparser.add_argument(
+        "-minh",
+        "--min-height",
+        dest="min_image_height",
+        default=400,
+        type=int,
+        help="Hauteur minimum de l'image à vérifier",
+    )
+    subparser.add_argument(
+        "-maxh",
+        "--max-height",
+        dest="max_image_height",
+        default=800,
+        type=int,
+        help="Hauteur maximum de l'image à vérifier",
+    )
+    subparser.add_argument(
+        "-r",
+        "--raise",
+        dest="raise_exceptions",
+        action="store_true",
+        default=False,
+        help="Lever des exceptions et donc arrêter le programme si des erreurs sont rencontrées",
+    )
+    subparser.set_defaults(func=run)
+    return subparser
+
+
+# ############################################################################
+# ########## MAIN ################
+# ################################
+
+
+def check_author_md(author: str, folder: Path) -> bool:
+    if author == "Geotribu":
+        return True
+    formatted = author.translate(str.maketrans({"'": "", " ": "-"}))
+    p = os.path.join(folder, f"{unidecode(formatted.lower())}.md")
```

Comment:
C'est overkill non pour gérer l'encodage vu que tous les fichiers sont supposément en UTF-8 ?

```suggestion
    p = os.path.join(folder, f"{unidecode(formatted.lower())}.md")
```

Reply:
C'est surtout pour remplacer les accents, du style `ë` -> `e` ...etc

Reply:
L'objectif c'est d'avoir le nom du fichier auteur c'est ça ?
Si oui, alors autant utiliser le slugger qui est utilisé dans le hook du site qui gère les bloc auteurs : https://github.com/geotribu/website/blob/9fc8acade6d197322df39d42f11b63dfa4da2ea5/hooks/mkdocs/G006_authors_block.py#L15

Reply:
Bien vu !